src/main/java/org/fluentd/logger/sender/AsyncRawSocketSender.java

```diff
@@ -106,9 +112,14 @@ public boolean emit(String tag, Map<String, Object> data) {
     @Override
     public boolean emit(final String tag, final long timestamp, final Map<String, Object> data) {
         final RawSocketSender sender = this.sender;
-        senderTask.execute(new EmitRunnable(tag, data, sender, timestamp));
-
-        return this.isConnected() || reconnector.enableReconnection(System.currentTimeMillis());
+        try {
+            Future<Boolean> result = senderTask.submit(new EmitRunnable(tag, data, sender, timestamp));
+            return result.get();
```

Comment:
Unfortunately, `Future#get` will wait the result here. So this method doesn't work asynchronously...
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get--